### PR TITLE
Hammerhead Interview

### PR DESCRIPTION
**Pull request by lekyle**
Used IntelliJ IDEA to compile and run program. Runs from main.kt

### CODE REVIEW

File: HammerheadInterview/src/Service.kt [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    {
+        val routeList = mutableListOf<String>()
+        var dest = ""
+        for(i in 1..4)
+        {
+            when(service)
+            {
+                "Strava" -> dest = id.toString() + routes.random()
+                "RWGPS" -> dest = routes.random() + id.toString()
+                "Komoot" -> dest = id.toString() + routes.random() + id.toString()
```
Comment (brentnd) [inline HammerheadInterview/src/Service.kt]: How would you refactor this if we didn't want to modify `Service` class every time we add a new service?

File: HammerheadInterview/src/Services.kt [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import java.lang.IllegalArgumentException
+
+class Services : IServices
```
Comment (brentnd) [inline HammerheadInterview/src/Services.kt]: How would you refactor this if we want the available services to be dynamic?

Comment (lekyle) [reply HammerheadInterview/src/Services.kt]: I resubmitted code with the refactoring changes requested.

**Comment by brentnd** [comment]
@lekyle thanks for the submission and the follow up to the review.

Closing this and we will follow up with you.

**Pull request closed without merging**